relay-event-schema/src/protocol/contexts/trace.rs

```diff
@@ -118,11 +118,58 @@ pub struct TraceContext {
     /// This flag only applies to events with [`Error`] type that have an associated dynamic sampling context.
     pub sampled: Annotated<bool>,
 
+    /// Arbitrary additional data on a trace.
+    #[metastructure(pii = "maybe", skip_serialization = "empty")]
+    pub data: Annotated<Data>,
+
     /// Additional arbitrary fields for forwards compatibility.
     #[metastructure(additional_properties, retain = "true", pii = "maybe")]
     pub other: Object<Value>,
 }
 
+/// The arbitrary data on the trace.
+#[derive(Clone, Debug, Default, PartialEq, Empty, FromValue, IntoValue, ProcessValue)]
+#[cfg_attr(feature = "jsonschema", derive(JsonSchema))]
+pub struct Data {
+    /// The current route in the application.
+    ///
+    /// Set by React Native SDK.
+    #[metastructure(pii = "maybe", skip_serialization = "empty")]
+    pub route: Annotated<Route>,
+    /// The previous route in the application
+    ///
+    /// Set by React Native SDK.
+    #[metastructure(field = "previousRoute", pii = "maybe", skip_serialization = "empty")]
+    pub previous_route: Annotated<Route>,
+
+    /// Additional arbitrary fields for forwards compatibility.
+    #[metastructure(
+        additional_properties,
+        retain = "true",
+        pii = "maybe",
+        skip_serialization = "empty"
+    )]
+    pub other: Object<Value>,
+}
+
+/// The route in the application, set by React Native SDK.
+#[derive(Clone, Debug, Default, PartialEq, Empty, FromValue, IntoValue, ProcessValue)]
+#[cfg_attr(feature = "jsonschema", derive(JsonSchema))]
+pub struct Route {
+    /// Parameters assigned to this route.
+    #[metastructure(pii = "true", skip_serialization = "empty", bag_size = "medium")]
+    params: Annotated<Object<Value>>,
```

Comment:
Not 100% sure if a `pii=true` child of a `pii=maybe` parent gets scrubbed automatically (I hope so). Should we add a test to `relay-pii` for an example input like `{"route": {"params": {"password": "pw-leaked"}}}`.

Reply:
yep, added test in [a325880](https://github.com/getsentry/relay/pull/2899/commits/a325880608e6552739f796d6889a67e36a3a2a1a), seems like scrubbing is working by default as expected.